types/geojson/index.d.ts

```diff
@@ -128,18 +130,24 @@ export interface MultiPolygon extends GeometryObject {
  * Geometry Collection
  * https://tools.ietf.org/html/rfc7946#section-3.1.8
  */
-export interface GeometryCollection extends GeometryObject {
+export interface GeometryCollection extends BaseGeometryObject {
     type: "GeometryCollection";
     geometries: Array<Point | LineString | Polygon | MultiPoint | MultiLineString | MultiPolygon>;
 }
 
+/**
+ * A geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3
+ */
+export type GeometryObject = GeometryCollection | LineString | MultiLineString | MultiPoint | MultiPolygon | Point | Polygon;
```

Comment:
We had a similar discussion [in the past](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/14299/files#diff-0747e8610da01ead57eb7491739cfa95R31), but there was also a problem with `GeometryCollections`. In spite of all we removed it later again.

The solution at the moment has no conflict with the problem we have to solve in the past. In addition don't have a need for something like a `BaseGeometryObject` (or a `DirectGeometryObject`) with the current solution (keep in mind that you don't have to take care with extending like you had in your `d3-geo` type definition in this PR!).

On the other hand your type definition is more precise.

@Cobster, @JeffJacobson what are your opinions on that?

Reply:
I'm afraid I'm not sure what `GeometryCollections` problem you are referring to, sorry. Is this something we can add a test for?